unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -1725,6 +1731,77 @@ inputDescription input =
       -- just trying to recover the syntax the user wrote
       These path _branch -> pure (Path.toText' path)
 
+handleStructuredFindI :: HQ.HashQualified Name -> Cli ()
+handleStructuredFindI rule = do
+  Cli.Env {codebase} <- ask
+  (ppe, names, rules0) <- lookupRewrite InvalidStructuredFind rule
+  let rules = snd <$> rules0
+  let fqppe = PPED.unsuffixifiedPPE ppe
+  results :: [(HQ.HashQualified Name, Referent)] <- pure $ do
+    r <- Set.toList (Relation.ran $ Names.terms (NamesWithHistory.currentNames names))
+    Just hq <- [PPE.terms fqppe r]
+    fullName <- [HQ'.toName hq]
+    guard (not (Name.beginsWithSegment fullName Name.libSegment))
```

Comment:
From the look of it this could still find names that _contain_ a lib segment; most of the time lib is top-level, but in scratch files you can run a find from anywhere.